core/src/runtime/mod.rs

```diff
@@ -852,10 +862,15 @@ impl Runtime {
         // Update the clk to the next cycle.
         self.state.clk += 4;
 
+        let channel = self.channel();
+        // Update the channel to the next cycle.
+        self.state.channel += 1;
+        self.state.channel %= NUM_BYTE_LOOKUP_CHANNELS;
```

Comment:
nit: Maybe collapse the two lines into one.

```suggestion
        self.state.channel = (self.state.channel + 1) % NUM_BYTE_LOOKUP_CHANNELS;
```